core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -104,12 +94,20 @@ object TestUtils extends Logging {
    * Create a temporary relative directory
    */
   def tempRelativeDir(parent: String): File = {
-    val f = new File(parent, "kafka-" + random.nextInt(1000000))
-    f.mkdirs()
+    new File(parent).mkdirs()
+    val f: File = Iterator.continually(new File(parent, "kafka-" + random.nextInt(1000000))).find(_.mkdir()).get
```

Comment:
Also, we may want to stop trying to create the dir after a number of attempts (it could fail due to permissions, file space or some other such thing). That could be done by tweaking the statement a little, example:

`Iterator.continually(new File(parent, "kafka-" + random.nextInt(1000000))).take(attempts).find(_.mkdir()).getOrElse(sys.error("Failed to create directory after $attempts")`